substratevm/src/com.oracle.svm.core.graal.llvm/src/com/oracle/svm/core/graal/llvm/LLVMGenerator.java

```diff
@@ -1469,24 +1469,36 @@ public Value emitXorFP(Value a, Value b) {
             throw unimplemented("the LLVM backend only supports XOR of integers, vectors and floating point numbers");
         }
 
+        private LLVMValueRef actualShiftingDistance(LLVMValueRef a, LLVMValueRef b) {
+            // https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.19
+
+            LLVMTypeRef typeA = typeOf(a);
+            int bitWidthA = LLVMIRBuilder.integerTypeWidth(typeA);
+            assert bitWidthA == 32 || bitWidthA == 64;
+
+            LLVMValueRef shiftDistanceBitMask = builder.constantInteger((bitWidthA == 32) ? 0x1f : 0x3f, bitWidthA);
```

Comment:
I'd prefer `bitWidthA - 1`, as it does not rely on the assertion being checked to be correct.

Reply:
Done.